### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by evgenii-komornyi**
- [x] Code coverage 87.9%.

### CODE REVIEW

File: Tests/BoardTest.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+                    Specie = "Test",
+                    Letter = ConsoleKey.T,
+                    Color = ConsoleColor.White,
+                    IsActive = true
+                }
+            };
+        }
+
+        [Fact]
+        public void CreateArray_CheckArrayLength_ReturnsHundredFifty()
```
Comment (ArvisZalitis) [inline Tests/BoardTest.cs:38]: Comment for all tests, missing AAA, for example
image

**Review by ArvisZalitis — APPROVED** [review]

File: Tests/BoardTest.cs (hunk 1 above)
Comment (evgenii-komornyi) [reply Tests/BoardTest.cs:38]: Done.

File: AnimalsTypes/Antilope.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+            Id = GenerateId();
+            Sex = (AnimalSex)Helper.random.Next(2);
+            IsPaired = false;
+            Vision = ConstantsRepository.AntilopeVision;
+            Health = ConstantsRepository.MaxHealth;
+            IsActive = true;
+            IsVisible = true;
+            Letter = ConsoleKey.A;
+            Color = ConsoleColor.Yellow;
+            Specie = ConstantsRepository.Antilope;
```
Comment (RaivisVitolins) [inline AnimalsTypes/Antilope.cs]:
```suggestion
            Species = ConstantsRepository.Antilope;
```
Singular for 'Species' is still 'Species'

File: AnimalsTypes/Antilope.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
+﻿using GameEngine.Entities;
+using GameEngine.Helpers;
+using Repository;
+
+namespace AnimalsTypes
```
Comment (RaivisVitolins) [inline AnimalsTypes/Antilope.cs]:
```suggestion
namespace AnimalTypes
```
```suggestion
namespace TypesOfAnimals
```

File: AnimalsTypes/Fox.cs [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        /// <summary>
+        /// Fox entity.
+        /// </summary>
+        public Fox()
+        {
+            Id = GenerateId();
+            Sex = (AnimalSex)Helper.random.Next(2);
+            IsPaired = false;
+            Vision = ConstantsRepository.FoxVision;
+            Health = ConstantsRepository.MaxHealth;
```
Comment (RaivisVitolins) [inline AnimalsTypes/Fox.cs:21]: Same MaxHealth for different animals?

File: GameEngine/Entities/Animal.cs [hunk 5]
```diff
@@ -1,220 +1,61 @@
...
-        /// <param name="currentHealth">Current health.</param>
-        private void Die(Animal currentAnimal, double currentHealth)
-        {
-            if (currentHealth <= 0)
-            {
-                currentAnimal.IsDead = true;
-            }
-        }
+        /// <param name="healthToIncrease"></param>
+        /// <returns></returns>
```
Comment (RaivisVitolins) [inline GameEngine/Entities/Animal.cs]: Either specify what is returned or remove tags as they are redundant.

File: GameEngine/Entities/Animal.cs [hunk 6]
```diff
@@ -1,220 +1,61 @@
...
-        /// <param name="currentAnimal">Current animal.</param>
-        /// <param name="currentHealth">Current health.</param>
-        private void Die(Animal currentAnimal, double currentHealth)
-        {
-            if (currentHealth <= 0)
-            {
-                currentAnimal.IsDead = true;
-            }
-        }
+        /// <param name="healthToIncrease"></param>
```
Comment (RaivisVitolins) [inline GameEngine/Entities/Animal.cs]: param missing description

File: GameEngine/Entities/Animal.cs [hunk 7]
```diff
@@ -1,220 +1,61 @@
...
-        /// <returns>Free cells to move.</returns>
-        protected List<NewAnimalCoordinates> CalculateCorrectPosition(Board board, List<Animal> animals, Animal animal)
-        {
-            List<NewAnimalCoordinates> freeCellsToMove = new List<NewAnimalCoordinates>();
-            NewAnimalCoordinates newCoordinates;
-
-            for (int newXCoordinate = animal.CoordinateX - 1; newXCoordinate <= animal.CoordinateX + 1; newXCoordinate++)
+        /// <param name="damage">Damage.</param>
+        /// <returns>Health after decreasing.</returns>
+        protected virtual double GetDamage(double damage) {
```
Comment (RaivisVitolins) [inline GameEngine/Entities/Animal.cs]:
```suggestion
        protected virtual double GetDamage(double damage)
        { 
```

File: GameEngine/Entities/Animal.cs [hunk 8]
```diff
@@ -1,220 +1,61 @@
...
-                currentAnimal.IsDead = true;
-            }
-        }
+        /// <param name="healthToIncrease"></param>
+        /// <returns></returns>
+        protected virtual double Heal(double healthToIncrease) => Health += healthToIncrease;
 
         /// <summary>
-        /// Calculates a new position based on the free cells around animal.
+        /// Decreases heals by damage.
```
Comment (RaivisVitolins) [inline GameEngine/Entities/Animal.cs]: 'heals' or health?

File: AnimalsTypes/Fox.cs (hunk 4 above)
Comment (evgenii-komornyi) [reply AnimalsTypes/Fox.cs:21]: For this task it was not described. I thouught that in this case Max health would be for all animals the same. But when other programmer will create new animal, he can define any max health for his animal.

File: AnimalsTypes/Antilope.cs (hunk 2 above)
Comment (evgenii-komornyi) [reply AnimalsTypes/Antilope.cs]: Done.

File: AnimalsTypes/Antilope.cs (hunk 3 above)
Comment (evgenii-komornyi) [reply AnimalsTypes/Antilope.cs]: Done.

File: GameEngine/Entities/Animal.cs (hunk 5 above)
Comment (evgenii-komornyi) [reply GameEngine/Entities/Animal.cs]: Done.

File: GameEngine/Entities/Animal.cs (hunk 6 above)
Comment (evgenii-komornyi) [reply GameEngine/Entities/Animal.cs]: Done.

File: GameEngine/Entities/Animal.cs (hunk 7 above)
Comment (evgenii-komornyi) [reply GameEngine/Entities/Animal.cs]: Done.

File: GameEngine/Entities/Animal.cs (hunk 8 above)
Comment (evgenii-komornyi) [reply GameEngine/Entities/Animal.cs]: Done.

File: GameEngine/Helpers/Helper.cs [hunk 9]
```diff
@@ -0,0 +1,354 @@
...
+        /// <param name="animalsAround">Animals around.</param>
+        /// <returns>Items around by type.</returns>
+        public static List<T> FindItemsAroundByType<T>(List<Animal> animalsAround) where T : Animal
+        {
+            return animalsAround
+                .Where(animal => typeof(T).IsAssignableFrom(animal.GetType()) && animal.IsActive && animal.IsVisible)
+                .Select(animal => (T)animal)
+                .ToList();
+        }
+
```
Comment (RaivisVitolins) [inline GameEngine/Helpers/Helper.cs]: 1 empty line is enough

**Review by RaivisVitolins — APPROVED** [review]

File: GameEngine/Helpers/Helper.cs (hunk 9 above)
Comment (evgenii-komornyi) [reply GameEngine/Helpers/Helper.cs]: Did not noticed. Thank you. Done.

**Pull request merged**